### ISSUE TITLE
[BUG]: Qt frontend should respect the XDG standard

### ISSUE THREAD

**Issue by j8r** [PCSX2/pcsx2#5599]
### Describe the Bug

When using the Qt frontend inside Flatpak, at first the `resources` directory, which is located at `/app/share/PCSX2/resources` inside the sandbox, is not found: `resources directory missing, your installation is incomplete`. I don't know how the wxWidgets frontend handle that. A quick fix is to move the folder where the binaries are.

After fixing that, `pcsx2-qt` tries to write directly to `$HOME/PCSX2`, which is forbidden by the Flatpak sandbox.

Finally, a `textures` folder is trying to be created in the same directory where the binaries are.

---

Command:
```sh
flatpak run --branch=master --command=pcsx2-qt net.pcsx2.PCSX2
```

Output (after fixing the `resources` directory error):
```sh
Qt: Session management error: Could not open network socket
Failed to save settings to '/home/j8r/PCSX2/inis/PCSX2.ini'.
BIOS Directory: /home/j8r/PCSX2/bios
Snapshots Directory: /home/j8r/PCSX2/snaps
Savestates Directory: /home/j8r/PCSX2/sstates
MemoryCards Directory: /home/j8r/PCSX2/memcards
Logs Directory: /home/j8r/PCSX2/logs
Cheats Directory: /home/j8r/PCSX2/cheats
CheatsWS Directory: /home/j8r/PCSX2/cheats_ws
Covers Directory: /home/j8r/PCSX2/covers
Game Settings Directory: /home/j8r/PCSX2/gamesettings
Cache Directory: /home/j8r/PCSX2/cache
Textures Directory: /app/bin/textures
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/home/j8r/PCSX2' couldn't be created (error 30: Read-only file system)
21:53:16: Error: Directory '/app/bin/textures' couldn't be created (error 30: Read-only file system)
```

---

Note: PCSX2 with the wxWidgets frontend works inside Flatpak, which respect `XDG_CONFIG_HOME`

### Reproduction Steps

Run the `pcsx2-qt` binary: `$HOME/PCSX2` is created no matter the `XDG_CONFIG_HOME` variables is, unlike [wxWidgets](https://github.com/PCSX2/pcsx2/blob/5a25cc171dd4ff6da59a09763342a55986624d54/pcsx2/gui/AppMain.cpp#L320).

Flatpak is not needed to reproduce and fix the issue, here is the related PR: https://github.com/flathub/net.pcsx2.PCSX2/pull/46.

### Expected Behavior

Even outside Flatpak it is not ideal to create a directory directly in the home. Generally applications write in `$HOME/.config`, and `XDG_CONFIG_HOME` should be taken if set.

The application can also read `XDG_DATA_DIRS` (and possibly others like `XDG_DATA_HOME`) environment variables. They are not always present depending of the environment, for example not in mine (Debian).

To sum-up the solution:
- Read `XDG_CONFIG_HOME`, fallback to `$HOME/.config`
- Do as wxWidgets to find the `resources`, or possibly read and parse the directories at `XDG_DATA_DIRS` to find the `resources` folder and create `textures` there.
On my machine, it looks like `/home/j8r/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share`

XDG Base Directory Specification: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables.

---

In my opinion data and configuration should be in separate places. `XDG_DATA_HOME` with `~/.local/share` as a fallback can be used (for example helps to version config files).

### PCSX2 Revision

v1.7.2412

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Debian

**Comment by stenzek** [comment]
Invalid, working as intended and I'm not going to change it. PRs welcome.

**Comment by RedDevilus** [comment]
As stated in earlier issues. We do not accept Qt GitHub issues at this time. PRs for improving it are fine however. Have a good day.

**Comment by stenzek** [comment]
The Qt frontend is also not ready for widespread use yet. It not working for packagers is a feature, not a bug, because you shouldn't be shipping an unfinished product, and creating support headaches for us.

**Comment by j8r** [comment]
Ok, got it. However this was not obvious. I haven't seen any warning during the build. Can this be stated in the issue template also? It will help for thoses that doesn't follow closely PCSX2 discussions nor issues.

I believe a special flag could be added to make it even clearer, some project do that (say `-DQT_BUILD_UNSUPPORTED_PREVIEW`)

**Comment by j8r** [comment]
This issue essentially duplicates #5496

**Comment by stenzek** [comment]
> However this was not obvious.

Do you see any Qt builds on the downloads page? That alone is pretty obvious.

> I believe a special flag could be added to make it even clearer

It's not necessary, the CI only builds Qt for Windows, and that's all we support.

Until someone steps up to maintain the interface for Linux, it will remain unsupported. So far, the only people who have been annoying about Qt have been packagers, who of course don't contribute, only complain.

**Comment by j8r** [comment]
> Do you see any Qt builds on the downloads page? That alone is pretty obvious.

Do you mean https://pcsx2.net/download/releases.html? The builds are quite out-of-date

Ha ok Qt is only supported for Windows now? A CI build does not necessarily always mean a tier 1 support, it just meana it compiles and perhaps ready to test for ealy adopters.

---

Anyway I can make a PR to add docs about that. Usually other projects make a table, Rust and Crystal for example, with each platform/architecture and the support tier.

Again the main issue for me at the end is not about the code and the status of the Qt frontend, but the doc. Me being dumb cannot be excluded, but others thought also like me by seing thr previous issues.

**Comment by j8r** [comment]
I want to clarify again, that's mainly for you to avoid new issues related to Qt landing here.

A bold text in the issue template sounds fair, what do you think?

**Comment by stenzek** [comment]
> Ha ok Qt is only supported for Windows now? A CI build does not necessarily always mean a tier 1 support, it just meana it compiles and perhaps ready to test for ealy adopters.

Bug reports for Qt are fine, considering I'm actioning them and still periodically working on bringing features across. There's just no Linux maintainer, because I don't know, apparently arguing is more important than bringing up support. I just refuse to work on a platform that I don't use; I'll make sure it builds, but that's all.

> Anyway I can make a PR to add docs about that. Usually other projects make a table, Rust and Crystal for example, with each platform/architecture and the support tier.

You're trying to create issues where there is none.
 - Linux users are a minority (low single digits),
 - Linux users who build from source (and don't use our provided appimages) are an even smaller minority again,
 - and Linux users who examine all the build options and may potentially enable a frontend which is disabled by default are pretty much non-existent.

The fact that no users have really pestered us about the state of Qt only proves my point. The only people who are, seem to be packagers who are for some reason trying to get ahead of the official builds, and ship something that is incomplete.

**Comment by j8r** [comment]
I was editing my message to clarify my point.
Packagers are not the enemy who only complain and pester, most people want to help by spending time to detail the issue in a constructive way - that's what I tried here. Issues are okay, really no big deal.

You have provided great work, I understand Linux is not a priority, the Qt frontend is WIP (got it now :slightly_smiling_face: ) and 
again here the issue is not about the code or maintainers (you rock, thanks!), at all - it is about documentation.
Until now I didn't even know PCSX2 were seeking maintainers for the Qt frontend!

It won't hurt to make it clear in the issue template and the README what this Qt frontend is and it is still WIP/not supported.
It can sound obvious after your detailed explanations, but not everyone have this directly in mind.

Here are examples on other projects:
- https://doc.rust-lang.org/nightly/rustc/platform-support.html
- https://crystal-lang.org/reference/1.3/platform_support.html

I propose something like in the README/issue template **The Qt frontend is NOT supported on Linux, maintainers welcome - no issue is accepted**.

Another option is to still accept issues but tag as "wont do", or "help welcome" to make it clear that no core member will fix it.

I tried to help, I hope this time taken will save more in the future.
